pkg/tfbridge/schema.go

```diff
@@ -1722,9 +1719,9 @@ func extractSchemaInputs(
 		return resource.NewArrayProperty(v)
 	case state.IsObject():
 		obj := state.ObjectValue()
-		if tfflds, ok := shimutil.CastToTypeObject(tfs); ok {
+		if tfflds, ok := tfs.Elem().(shim.Resource); ok {
```

Comment:
What happens here is `tfs` does not have a valid Type() because it came from elemSchemas, therefore it fails to pass the CastToTypeObject TypeMap check.